### PR TITLE
Remove old code in UnPickler

### PR DESCRIPTION
**Pull request by diesalbla**

### CODE REVIEW

File: src/reflect/scala/reflect/internal/pickling/UnPickler.scala [hunk 1]
```diff
@@ -721,7 +721,7 @@ abstract class UnPickler {
       private[this] val p = phase
       protected def completeInternal(sym: Symbol) : Unit = try {
         completingStack += sym
-        val tp = at(i, () => readType(sym.isTerm)) // after NMT_TRANSITION, revert `() => readType(sym.isTerm)` to `readType`
+        val tp = at(i, () => readType()) // after NMT_TRANSITION, revert `() => readType()` to `readType`
```
Comment (hrhino) [inline src/reflect/scala/reflect/internal/pickling/UnPickler.scala]: I think you've done the TODO.

Also, you could save a few cycles by removing the temporary fix below.

Comment (diesalbla) [reply src/reflect/scala/reflect/internal/pickling/UnPickler.scala]: I was asking earlier today if anyone knew what that TODO meant. https://gitter.im/scala/contributors?at=5dbdcf609825bd6bac14ff6e

Comment (diesalbla) [reply src/reflect/scala/reflect/internal/pickling/UnPickler.scala]: As to the temporary fix, it seems @lrytz was the author, so he may advise on the matter (or remove it himself).

Comment (hrhino) [reply src/reflect/scala/reflect/internal/pickling/UnPickler.scala]: NMT is `NullaryMethodType`, which didn't always exist, apparently.

Comment (lrytz) [reply src/reflect/scala/reflect/internal/pickling/UnPickler.scala]: Indeed (f0bff86d31). The commend can be removed, `at(i, readType)`.

The fix below can also go.

Comment (diesalbla) [reply src/reflect/scala/reflect/internal/pickling/UnPickler.scala]: With "the fix below gan also go", are you referring to lines 726 to 738, and that `fixLocalChildTp` can simply be replaced with `tp`?

Comment (lrytz) [reply src/reflect/scala/reflect/internal/pickling/UnPickler.scala]: Yes, remove `fixLocalChildTp` and use `tp` instead.

File: src/reflect/scala/reflect/internal/pickling/UnPickler.scala [hunk 2]
```diff
@@ -721,25 +721,13 @@ abstract class UnPickler {
...
-        // pickler did not add the ObjectTpe superclass, it used a trait as the first parent. This
-        // tripped an assertion in AddInterfaces which checks that the first parent is not a trait.
-        // This workaround can probably be removed in 2.12, because the 2.12 compiler is supposed
-        // to only read classfiles generated by 2.12.
-        val fixLocalChildTp = if (sym.rawname == tpnme.LOCAL_CHILD) tp match {
-            case ClassInfoType(superClass :: traits, decls, typeSymbol) if superClass.typeSymbol.isTrait =>
-              ClassInfoType(definitions.ObjectTpe :: superClass :: traits, decls, typeSymbol)
-            case _ => tp
-          } else tp
+        val tp = at(i, () => readType())
```
Comment (hrhino) [inline src/reflect/scala/reflect/internal/pickling/UnPickler.scala:724]: `at(i, readType)` is almost fluent.

**Review by hrhino — APPROVED** [review]
Nice!

**Review by retronym — APPROVED** [review]

**Comment by retronym** [comment]
/nothingtoseehere

(Travis build failure is a know issue fixed by #8528 )

**Pull request merged**